lib/OptionsDefaulter.js

```diff
@@ -27,44 +23,53 @@ function setProperty(obj, name, value) {
 	obj[name.pop()] = value;
 }
 
-OptionsDefaulter.prototype.process = function(options) {
-	var addItemTo = function addItemTo(list) {
-		return function(item) {
-			list.push(item);
-		}
+class OptionsDefaulter {
+	constructor() {
+		this.defaults = {}
+		this.config = {};
 	}
 
-	for(var name in this.defaults) {
-		switch(this.config[name]) {
-			case undefined:
-				if(getProperty(options, name) === undefined)
-					setProperty(options, name, this.defaults[name]);
-				break;
-			case "call":
-				setProperty(options, name, this.defaults[name].call(this, getProperty(options, name), options), options);
-				break;
-			case "make":
-				if(getProperty(options, name) === undefined)
-					setProperty(options, name, this.defaults[name].call(this, options), options);
-				break;
-			case "append":
-				var oldValue = getProperty(options, name);
-				if(!Array.isArray(oldValue)) oldValue = [];
-				this.defaults[name].forEach(addItemTo(oldValue));
-				setProperty(options, name, oldValue);
-				break;
-			default:
-				throw new Error("OptionsDefaulter cannot process " + this.config[name]);
+	process(options) {
+		const addItemTo = function addItemTo(list) {
+			return function(item) {
+				list.push(item);
+			}
+		}
+
+		for(let name in this.defaults) {
+			switch(this.config[name]) {
+				case undefined:
+					if(getProperty(options, name) === undefined)
+						setProperty(options, name, this.defaults[name]);
+					break;
+				case "call":
+					setProperty(options, name, this.defaults[name].call(this, getProperty(options, name), options), options);
+					break;
+				case "make":
+					if(getProperty(options, name) === undefined)
+						setProperty(options, name, this.defaults[name].call(this, options), options);
+					break;
+				case "append":
+					let oldValue = getProperty(options, name);
+					if(!Array.isArray(oldValue)) oldValue = [];
+					this.defaults[name].forEach(addItemTo(oldValue));
```

Comment:
Actually, given that this is the only use of `addItemTo`, it can be deleted and this line replaced with `oldValue.push.apply(oldValue, this.defaults[name])` (mutates oldValue), or `[].concat(oldValue, this.defaults[name])` (pure)